ddl/partition.go

```diff
@@ -163,32 +163,114 @@ func checkPartitionNameUnique(tbInfo *model.TableInfo, pi *model.PartitionInfo)
 	return nil
 }
 
+// For partition tables, mysql do not support Constant, random or timezone-dependent expressions
+// Based on mysql code to check whether field is valid, every time related type has check_valid_arguments_processor function.
+// See https://github.com/mysql/mysql-server/blob/5.7/sql/item_timefunc
+type isTimezoneDependentProcessor func(columns []*expression.Column) bool
+
+// See https://github.com/mysql/mysql-server/blob/5.7/sql/item_func.h#L387
+func hasTimestampField(columns []*expression.Column) bool {
+	for _, c := range columns {
+		if c.GetType().Tp == mysql.TypeTimestamp {
+			return true
+		}
+	}
+
+	return false
+}
+
+// See https://github.com/mysql/mysql-server/blob/5.7/sql/item_func.h#L399
+func hasDateField(columns []*expression.Column) bool {
+	for _, c := range columns {
+		if c.GetType().Tp == mysql.TypeDate || c.GetType().Tp == mysql.TypeDatetime {
+			return true
+		}
+	}
+
+	return false
+}
+
+// See https://github.com/mysql/mysql-server/blob/5.7/sql/item_func.h#L412
+func hasTimeField(columns []*expression.Column) bool {
+	for _, c := range columns {
+		if c.GetType().Tp == mysql.TypeDatetime {
```

Comment:
`if c.GetType().Tp == mysql.TypeDatetime || c.GetType().Tp == mysql.TypeDuration {`
/* TypeDuration original name was TypeTime, renamed to TypeDuration to resolve the conflict with Go type Time.*/

Reply:
Got it, thanks.